### DISCUSSION TITLE
`default` as a function?

### DISCUSSION THREAD

**Discussion by Juanc1to** [SBoudrias/Inquirer.js#1147] [Q&A]
[The documentation indicates that the `default` property on a question object can be a function](https://github.com/SBoudrias/Inquirer.js#questions), but I don't see any support for that in the code base or the test cases.  Is that a mistake of the documentation?  Would that feature even be particularly useful?

One reason that I'm asking is that I'm doing a bit of work on a plugin, and I'm curious about providing full support for the `default` property so that it works as users would expect.

**Comment by SBoudrias** [comment]
Hey 👋🏻 

Would you be able to provide more context as to which code you looked at? Also which package are you using? `inquirer` or the beta `@inquirer/*`?

**Reply by Juanc1to** [reply]
Hi Simon, thanks for responding and nice to meet you!  I'm using `inquirer`.  (On what branch is development on `@inquirer/*` happening?)  As an example of what I'm seeing, the `InputPrompt` class uses the `default` question property here: <https://github.com/SBoudrias/Inquirer.js/blob/edc49ce0c8c54872f08d4616545450955acbb657/packages/inquirer/lib/prompts/input.js#L75>
Note that it simply references the value, instead of checking to see if it's a function.

**Reply by Juanc1to** [reply]
Ah, but now I see that in my ignorance I had not looked at how the driver code functions.  I see that `PromptUI.processQuestion` uses `utils.fetchAsyncQuestionProperty` to replace the `default` property with the result of calling the function.  Is there a test that covers these function properties?

**Reply by SBoudrias** [reply]
> Is there a test that covers these function properties?

Quite honestly this is hard to answer... Most of `inquirer` was written ~10 years ago. Best way to know is to remove/comment the feature and run `yarn test`.
